SourcePointMetaApp/SourcePointMetaAppTests/ConsentViewControllerTests/ConsentViewControllerTest.swift

```diff
@@ -0,0 +1,295 @@
+//
+//  ConsentViewControllerTest.swift
+//  ConsentViewControllerTests
+//
+//  Created by Vilas on 8/29/19.
+//
+
+import XCTest
+@testable import ConsentViewController
+
+class ConsentViewControllerTest: XCTestCase, ConsentDelegate {
+    var consentViewController: ConsentViewController?
+    
+    var accountId: Int = 0
+    var siteId: Int = 0
+    var siteName: String = ""
+    var campaign: String = ""
+    var pmId: String = ""
+    var showPM: Bool = false
+    var authId: String = ""
+    var consentString: String = ""
+    
+    override func setUp() {
+        readDataFromPlist()
+        consentViewController = try! ConsentViewController(accountId: accountId, siteId: siteId, siteName: siteName, PMId: pmId, campaign: campaign, showPM: showPM,consentDelegate: self)
+    }
+    
+    override func tearDown() {
+        
+    }
+    
+    /// this method is used to read the data from plist
+    func readDataFromPlist() {
+        if let path = Bundle(for: type(of: self)).path(forResource: "TestData", ofType: "plist") {
+            let testData = NSDictionary(contentsOfFile: path)
+            accountId = testData?.value(forKey: "AccountId") as! Int
+            siteId = testData?.value(forKey: "SiteId") as! Int
+            siteName = testData?.value(forKey: "SiteName") as! String
+            showPM = testData?.value(forKey: "ShowPM") as! Bool
+            pmId = testData?.value(forKey: "PMId") as! String
+            campaign = testData?.value(forKey: "Campaign") as! String
+            authId = testData?.value(forKey: "AuthId") as! String
+            consentString = testData?.value(forKey: "ConsentString") as! String
+        }
+    }
+    
+    /// this method is used to test whether webview is loaded or not
+    func testLoadView() {
+        consentViewController?.loadView()
+        if consentViewController?.webView != nil {
+            XCTAssert(true, "Webview initialized successfully")
+        } else {
+            XCTAssert(false, "Unable to initialize webview")
+        }
+    }
+    
+    /// this method is used to test whether message url is created or not successfully
+    func testGetMessageUrlWithAuthId() {
```

Comment:
I believe this belongs to `SourcePointClientTest` rather than `ConsentViewControllerTest`. 
Same for the function below `testGetMessageUrlWithoutAuthId`
Furthermore, we should assert upon its return, not only `!= nil`

Reply:
In `ConsentViewController` class also we have methods where we are getting message URL.
We can use dynamic data for testing from `TestData.plist` file. So If I used assert upon return value and used account/property is different then test case will get fail.

Reply:
I have added assert on real value where it is possible